packages/pwa-kit-react-sdk/src/ssr/universal/components/switch/index.jsx

```diff
@@ -20,17 +20,20 @@ import {UIDReset, UIDFork} from 'react-uid'
  * @private
  */
 const Switch = (props) => {
-    const {error, appState, routes, App} = props
+    const {error, appState, routes, App, locale} = props
     return (
         <UIDReset>
             <AppErrorBoundary error={error}>
                 {!error && (
                     <App preloadedProps={appState.appProps}>
                         <RouterSwitch>
                             {routes.map((route, i) => {
-                                const {component: Component, ...routeProps} = route
+                                const {component: Component, path, localized, ...routeProps} = route
+
+                                let localizedPath = localized ? `/${locale}${path}` : path
```

Comment:
this could be extracted into a static method of Switch.

Reply:
the format of the localized path is not customizable, a possible solution is to take a callback function (that takes locale as input and localizedPath as output) from user land.